ipn/ipnlocal/local.go

```diff
@@ -612,12 +612,35 @@ func (b *LocalBackend) Start(opts ipn.Options) error {
 		b.logf("Start")
 	}
 
+	b.mu.Lock()
+
+	// The iOS client sends a "Start" whenever it UI screen comes
+	// up, just because it wants a netmap. That should be fixed,
+	// but meanwhile we can make Start cheaper here for such a
+	// case and not restart the world (which takes a few seconds).
+	// Instead, just send a notify with the state that iOS needs.
+	if b.state == ipn.Running &&
+		opts.Prefs == nil &&
+		opts.AuthKey == "" &&
+		b.stateKey == opts.StateKey &&
+		b.hostinfo != nil &&
+		b.hostinfo.FrontendLogID == opts.FrontendLogID {
```

Comment:
Can we encapsulate this check somewhere?

Reply:
Done